Azure-Samples/app-service-javascript-sap-cloud-sdk-quickstart
# enhanced error handling for bp controller

## Purpose

This PR contains:

* Adjustment of error handling for health endpoint and business partner controller
* Introduction of logging via SAP Cloud SDK logger

## Does this introduce a breaking change?

```
[X] Yes - Error message content potentially changed
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

```
- Execute the defined calls to a S/4HANA system with a missing/wrong API key to check the error handling
- Execute the defined calls to a S/4HANA system with a correct API key to check that the flow still works
```

## What to Check
Verify that the following are valid

* Calls should work, error cases should return consistent error messages

## Other Information

The Azure APIM setup could not be tested. We probably need an extension of the switch in `reateHttpException` of the `business-partner.controller.ts` if we now the exact type of the exception. If it is a generic exception we might need to another way to identify the type or we enhance the generic creation of an HTTP exception.
